### PR TITLE
added OrderController, AuthenticationController, UserController, Shop…

### PR DESCRIPTION
**Pull request by Fedich09**
…pingCartController and implementation methods; added all helpers class like ResponseDto, RequestDto and Mapper for classes also

### CODE REVIEW

**Review by MykolaFatkullin — APPROVED** [review]

File: src/main/java/com/dev/cinema/controller/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        this.userService = userService;
+        this.cartService = cartService;
+    }
+
+    @PostMapping("/complete")
+    public void completeOrder(@RequestParam Long userId) {
+        orderService.completeOrder(cartService.getByUser(userService.get(userId)));
+    }
+
+    @RequestMapping
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controller/OrderController.java]: @GetMapping?

File: src/main/java/com/dev/cinema/controller/ShoppingCartController.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        this.movieSessionService = movieSessionService;
+        this.userService = userService;
+    }
+
+    @PostMapping("/movie-sessions")
+    public void addMovieSession(@RequestParam Long userId, @RequestParam Long movieSessionId) {
+        cartService.addSession(movieSessionService.get(movieSessionId), userService.get(userId));
+    }
+
+    @RequestMapping("/by-user")
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controller/ShoppingCartController.java]: same

File: src/main/java/com/dev/cinema/controller/UserController.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+public class UserController {
+    private final UserService userService;
+    private final UserMapper userMapper;
+
+    public UserController(UserService userService, UserMapper userMapper) {
+        this.userService = userService;
+        this.userMapper = userMapper;
+    }
+
+    @RequestMapping("/by-email")
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controller/UserController.java]: same

File: src/main/java/com/dev/cinema/dao/UserDao.java [hunk 4]
```diff
@@ -7,4 +7,6 @@
     User add(User user);
 
     Optional<User> findByEmail(String email);
+
+    User get(Long id);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/dao/UserDao.java]: should we return Optional instead?

File: src/main/java/com/dev/cinema/model/dto/authentication/AuthenticationRequestDto.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
+package com.dev.cinema.model.dto.authentication;
+
+import org.springframework.stereotype.Component;
+
+@Component
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/dto/authentication/AuthenticationRequestDto.java]: why @Component?

File: src/main/java/com/dev/cinema/model/dto/order/OrderResponseDto.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
+package com.dev.cinema.model.dto.order;
+
+import java.time.LocalDate;
+import java.util.List;
+
+public class OrderResponseDto {
+    private Long id;
+    private List<Long> ticketIds;
+    private LocalDate orderDateTime;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/dto/order/OrderResponseDto.java]:
```suggestion
    private String orderDateTime;
```

File: src/main/java/com/dev/cinema/model/dto/shoppingcart/ShoppingCartResponseDto.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
+package com.dev.cinema.model.dto.shoppingcart;
+
+import java.util.List;
+
+public class ShoppingCartResponseDto {
+    private Long id;
+    private List<Long> ticketsIds;
+    private Long userId;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/dto/shoppingcart/ShoppingCartResponseDto.java]: please see checklist

File: src/main/java/com/dev/cinema/model/dto/shoppingcart/ShoppingCartResponseDto.java [hunk 8]
```diff
@@ -0,0 +1,33 @@
+package com.dev.cinema.model.dto.shoppingcart;
+
+import java.util.List;
+
+public class ShoppingCartResponseDto {
+    private Long id;
+    private List<Long> ticketsIds;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/dto/shoppingcart/ShoppingCartResponseDto.java]:
```suggestion
    private List<Long> ticketIds;
```

File: src/main/java/com/dev/cinema/service/mapper/AuthenticationMapper.java [hunk 9]
```diff
@@ -0,0 +1,9 @@
+package com.dev.cinema.service.mapper;
+
+import com.dev.cinema.model.User;
+import com.dev.cinema.model.dto.authentication.AuthenticationRequestDto;
+import com.dev.cinema.service.mapper.genericmapper.GenericMapperRequest;
+
+public interface AuthenticationMapper extends GenericMapperRequest<User,
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/mapper/AuthenticationMapper.java]: do we use this mapper?

File: src/main/java/com/dev/cinema/service/mapper/genericmapper/GenericMapperRequest.java [hunk 10]
```diff
@@ -0,0 +1,5 @@
+package com.dev.cinema.service.mapper.genericmapper;
+
+public interface GenericMapperRequest<T, V> {
+    T toEntity(V object);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/mapper/genericmapper/GenericMapperRequest.java]:
```suggestion
    T toEntity(V requestDto);
```

File: src/main/java/com/dev/cinema/service/mapper/genericmapper/GenericMapperResponse.java [hunk 11]
```diff
@@ -0,0 +1,5 @@
+package com.dev.cinema.service.mapper.genericmapper;
+
+public interface GenericMapperResponse<T, V> {
+    T toDto(V object);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/mapper/genericmapper/GenericMapperResponse.java]:
```suggestion
    T toDto(V entity);
```

File: src/main/java/com/dev/cinema/service/mapper/impl/OrderMapperImpl.java [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+public class OrderMapperImpl implements OrderMapper {
+    @Override
+    public OrderResponseDto toDto(Order order) {
+        OrderResponseDto orderResponseDto = new OrderResponseDto();
+        orderResponseDto.setId(order.getId());
+        orderResponseDto.setOrderDateTime(order.getOrderDate().toLocalDate());
+        orderResponseDto.setTicketIds(order.getTickets().stream()
+                .map(Ticket::getId)
+                .collect(Collectors.toList()));
+        return orderResponseDto;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/mapper/impl/OrderMapperImpl.java:21]: what about userId?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/service/mapper/impl/OrderMapperImpl.java (hunk 12 above)
Comment (Fedich09) [reply src/main/java/com/dev/cinema/service/mapper/impl/OrderMapperImpl.java:20]: cartId and userId hold the same value

File: src/main/java/com/dev/cinema/service/mapper/AuthenticationMapper.java (hunk 9 above)
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/mapper/AuthenticationMapper.java]: let's remove this interface as well

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**